Orange/data/table/base.py

```diff
@@ -0,0 +1,1292 @@
+        # (subclass) type
+        if cls != data.__class__:
+            data = cls(data)
+
+        data.name = os.path.splitext(os.path.split(filename)[-1])[0]
+        data.__file__ = absolute_filename
+        return data
+
+    @classmethod
+    def from_url(cls, url):
+        """Read a table from a URL.
+
+        Parameters
+        ----------
+        url : str
+            The URL to read the table from.
+
+        Returns
+        -------
+        TableBase
+            A table constructed from the given URL.
+        """
+        from Orange.data.io import UrlReader
+        reader = UrlReader(url)
+        data = reader.read()
+        if cls != data.__class__:
+            data = cls(data)
+        return data
+
+    def _to_numpy(self, X=False, Y=False, meta=False, writable=False):
+        """Export a transformed numpy matrix.
+
+        The order is always X, Y, meta. Always 2D.
+        The columns are in the same order as in Table.domain._.
+        If writable == False (default), the numpy writable flag is set to false.
+        This means write operations on this array will loudly fail.
+
+        This is not the same as .values, but instead transforms the attributes to
+        numeric values (where possible); e.g. uses var.values indices instead of
+        actual descriptors.
+
+        Parameters
+        ----------
+        X : bool, default False
+            Whether to include the domain attributes in the result.
+        Y : bool, default False
+            Whether to include the domain class variables in the result.
+        meta : bool, default False
+            Whether to include the domain metas in the result.
+        writable : bool, default False
+            Whether to mark the resulting domain as writable.
+        Returns
+        -------
+        np.ndarray
+            The numpy array of the selected and transformed table data.
+        """
+        cols = []
+        cols += self.domain.attributes if X else []
+        cols += self.domain.class_vars if Y else []
+        cols += self.domain.metas if meta else []
+
+        # support using this in TableSeries, whose len gives the number of columns
+        n_rows = 1 if isinstance(self, SeriesBase) else len(self)
+
+        # preallocate result, we fill it in-place
+        # we need a more general dtype for metas (commonly strings),
+        # otherwise assignment fails later
+        result = np.zeros((n_rows, len(cols)), dtype=object if meta else None)
+        # effectively a double for loop, see if this is a bottleneck later
+        for i, col in enumerate(cols):
+            if isinstance(self, SeriesBase):
+                # if this is used in TableSeries, we don't have a series but an element
+                result[:, i] = col.to_val(self[col])
+            else:
+                vals = col.to_val(self[col]).values
+                result[:, i] = vals.values if hasattr(vals, 'values') else vals
```

Comment:
Huh, I'm sure I needed this for a test somewhere. Looking at all of them again, it works without this, weird.